Absoft compile error in opal shmem

@devreal We're getting a compile error on master because of c385c927fbfd6964d5a6167b7837ae2c1d969839.  Can you look at this?

Absoft MTT (see [here](https://mtt.open-mpi.org/index.php?do_redir=3193) and [here](https://mtt.open-mpi.org/index.php?do_redir=3194)) is running into a compile error on master:
 
```
In file included from ../../../opal/mca/shmem/shmem.h:52,
                 from base/shmem_base_open.c:29:
../../../opal/mca/shmem/shmem_types.h:107: error: requested alignment is not a power of 2
In file included from ../../../opal/mca/shmem/shmem.h:52,
                 from base/shmem_base_close.c:27:
../../../opal/mca/shmem/shmem_types.h:107: error: requested alignment is not a power of 2
In file included from ../../../opal/mca/shmem/shmem.h:52,
                 from base/shmem_base_wrappers.c:25:
../../../opal/mca/shmem/shmem_types.h:107: error: requested alignment is not a power of 2
make[2]: *** [base/shmem_base_open.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [base/shmem_base_close.lo] Error 1
make[2]: *** [base/shmem_base_wrappers.lo] Error 1
In file included from ../../../opal/mca/shmem/shmem.h:52,
                 from base/shmem_base_select.c:30:
../../../opal/mca/shmem/shmem_types.h:107: error: requested alignment is not a power of 2make[2]:
*** [base/shmem_base_select.lo] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all-recursive] Error 1
```

The problem appears to be:

https://github.com/open-mpi/ompi/blob/8343a289f227d4fb631354079e7521883a34ec30/opal/mca/shmem/shmem_types.h#L107

c385c927fbfd6964d5a6167b7837ae2c1d969839 was actually from many months ago; I don't know if there was a recent commit that changed the value of `OPAL_ALIGN_MIN`...?

@devreal can you work with our Absoft contact, Tony (@cagoelz) to figure out what's happening here?

--- a/opal/include/opal/align.h
+++ b/opal/include/opal/align.h
@@ -32,7 +32,15 @@
 #include <stddef.h>
 #define OPAL_ALIGN_MIN (_Alignof(max_align_t))
 #else
-#define OPAL_ALIGN_MIN (sizeof(long double))
+#if defined(OPAL_ALIGNMENT___FLOAT128)
+#define OPAL_ALIGN_MIN (OPAL_ALIGNMENT___FLOAT128)
+#elif defined(OPAL_ALIGNMENT_LONG_DOUBLE_COMPLEX)
+#define OPAL_ALIGN_MIN (OPAL_ALIGNMENT_LONG_DOUBLE_COMPLEX)
+#elif defined (OPAL_ALIGNMENT_LONG_DOUBLE)
+#define OPAL_ALIGN_MIN (OPAL_ALIGNMENT_LONG_DOUBLE)
+#else
+#define OPAL_ALIGN_MIN (OPAL_ALIGNMENT_DOUBLE)
+#endif
 #endif /* __STDC_VERSION__ >= 201101L */
 
 #endif /* OPAL_ALIGN_H */

